gno.land/pkg/sdk/vm/params.go

```diff
@@ -91,6 +104,97 @@ func (vm *VMKeeper) getSysNamesPkgParam(ctx sdk.Context) string {
 	return sysNamesPkg
 }
 
-func (vm *VMKeeper) WillSetParam(ctx sdk.Context, key string, value interface{}) {
-	// XXX validate input?
+func (vm *VMKeeper) getValsetRealmParam(ctx sdk.Context) string {
+	// Fetch the latest valset realm path
+	valsetRealm := ValsetRealmDefault
+	vm.prmk.GetString(ctx, ValsetRealmParamPath, &valsetRealm)
+
+	return valsetRealm
+}
+
+var reValsetUpdates = regexp.MustCompile(fmt.Sprintf("^%s_\\d+$", "valset_updates"))
+
+func (vm *VMKeeper) WillSetParam(ctx sdk.Context, key string, value any) {
+	switch {
+	// vm:<valset-realm-path>:valset_updates_<height>
+	case strings.HasPrefix(key, vm.getValsetRealmParam(ctx)+":"+"valset_updates"):
+		// This update is realm-specific, so we trim the realm part
+		parts := strings.Split(key, ":")
+		
+		if !reValsetUpdates.Match([]byte(parts[1])) {
+			panic(
+				fmt.Sprintf(
+					"invalid valset update key: %s",
+					key,
+				),
+			)
```

Comment:
```suggestion
			panic("invalid valset update key: " + key)
```

I know that you usually nitpick for the opposite, but I believe that six lines for this simple panic is a good example of when we should embrace simplicity and master the art of compromise. :)